Examples/Minesweeper/Minesweeper/State/MinesweeperState.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  MinesweeperState.swift
+//  Minesweeper
+//
+//  Copyright © 2016 Bending Spoons.
+//  Distributed under the MIT License.
+//  See the LICENSE file for more information.
+
+import Katana
+
+struct MinesweeperState: State {
+  var gameOver: Bool
+  let rows: Int
+  let cols: Int
+  var mines: [Bool]
+  var disclosed: [Bool]
+  
+  init() {
+    self.init(difficulty: .hard)
+  }
+  
+  init(difficulty: Difficulty) {
+    switch difficulty {
+    case .easy:
+      self.init(cols: 9, rows: 9, mines: 10)
+    case .medium:
+      self.init(cols: 15, rows: 16, mines: 35)
+    case .hard:
+      self.init(cols: 15, rows: 30, mines: 90)
+    }
+  }
+  
+  private init(cols: Int, rows: Int, mines: Int) {
+    self.gameOver = false
+    self.rows = rows
+    self.cols = cols
+    self.mines = Array(repeating: false, count: rows * cols)
+    self.disclosed = Array(repeating: false, count: rows * cols)
+    self.poseMines(numberOfMines: mines)
+  }
+  
+}
+
+
+// MARK: - Mines Operations
+extension MinesweeperState {
+  fileprivate mutating func poseMines(numberOfMines: Int) {
+    var i = 0
+    while i < numberOfMines {
+      let r = Int.random(max: rows)
+      let c = Int.random(max: cols)
+      if( self[c, r] != true) {
+        i += 1
+        self[c, r] = true
+      }
+    }
+  }
+  
+  func minesNearbyCellAt(col: Int, row: Int) -> Int {
```

Comment:
actually it's not possible because `minesNearbyCellAt` is used in the connect function of the minesweeperCell